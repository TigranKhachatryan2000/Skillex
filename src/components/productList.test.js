import React from 'react';
import { render, screen } from '@testing-library/react';

import ProductList from './productList';

describe('ProductList Component', () => {
  test('renders "No products found" message when there are no products', () => {
    const { asFragment } = render(<ProductList products={[]} />);

    expect(asFragment()).toMatchSnapshot();

    const noProductsMessage = screen.getByText(/no products found/i);
    expect(noProductsMessage).toBeInTheDocument();
  });

  test('renders a list of products', () => {
    const mockProducts = [
      { id: 1, name: 'Product 1', price: 99.99, rating: 4 },
      { id: 2, name: 'Product 2', price: 149.99, rating: 5 },
      { id: 3, name: 'Product 3', price: 49.99, rating: 3 },
    ];

    const { asFragment } = render(<ProductList products={mockProducts} />);

    expect(asFragment()).toMatchSnapshot();

    mockProducts.forEach((product) => {
      expect(screen.getByText(product.name)).toBeInTheDocument();
      expect(screen.getByText(`${product.rating} Stars`)).toBeInTheDocument();
      expect(screen.getByText(`$${product.price.toFixed(2)}`)).toBeInTheDocument();
    });
  });
});
