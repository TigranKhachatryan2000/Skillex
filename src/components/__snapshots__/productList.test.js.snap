// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component renders "No products found" message when there are no products 1`] = `
<DocumentFragment>
  <p>
    No products found
  </p>
</DocumentFragment>
`;

exports[`ProductList Component renders a list of products 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h3>
        Product 1
      </h3>
      <p>
        $99.99
      </p>
      <p>
        4 Stars
      </p>
    </div>
    <div>
      <h3>
        Product 2
      </h3>
      <p>
        $149.99
      </p>
      <p>
        5 Stars
      </p>
    </div>
    <div>
      <h3>
        Product 3
      </h3>
      <p>
        $49.99
      </p>
      <p>
        3 Stars
      </p>
    </div>
  </div>
</DocumentFragment>
`;
